ecosystem/indexer-grpc/indexer-grpc-table-info/src/runtime.rs

```diff
@@ -0,0 +1,85 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::table_info_parser::TableInfoParser;
+use aptos_api::context::Context;
+use aptos_config::config::NodeConfig;
+use aptos_logger::{error, info};
+use aptos_mempool::MempoolClientSender;
+use aptos_moving_average::MovingAverage;
+use aptos_storage_interface::DbReaderWriter;
+use aptos_types::chain_id::ChainId;
+use std::{env, sync::Arc};
+use tokio::runtime::Runtime;
+
+pub fn bootstrap(
+    config: &NodeConfig,
+    chain_id: ChainId,
+    db: DbReaderWriter,
+    mp_sender: MempoolClientSender,
+) -> Option<Runtime> {
+    if !config.storage.enable_indexer_async_v2 {
```

Comment:
actually this one should be its own config as well.

Reply:
why thou? we still need indexer async v2 rocksdb to be exposed for read & write through aptosdb.

Reply:
Because we're trying to get out of storage right? This whole thing was a hack on top of storage APIs but now we're splitting it out into its own thread so why share the same config?

Reply:
To put it simply, I see that every runtime has its own config. It's nice a clean. Why break this paradigm?

Reply:
i totally agree the paradigm, reason why im doing this way instead of the way you suggested is purely implementation challenge. lets take this example: this [fn](https://github.com/aptos-labs/aptos-core/blob/03faff98fcf6e8fafcbaa8b9290f5d3ccdd1fc13/api/types/src/convert.rs#L418-L446) is trying to convert table item to table info data. it gets table info data from this aptosdb exposed read [api](https://github.com/aptos-labs/aptos-core/blob/ccfffc8ea405743ce88165908ddc2d20439effbe/storage/aptosdb/src/lib.rs#L2177-L2182), because it uses move_converter to convert from bcs to json. now in our scenario, we still want apis to read table info, right? so `indexer async v2` will still be part of the aptosdb. that being said, we could 100% remove the config from `StorageConfig`, but `indexer async v2` db will be opened in aptosdb so that it could be exposed for read and write to other services in the fullnode if that makes sense.

Reply:
What different does it make if you call config.storage.enable... vs config.table_info.enable? It only needs to import NodeConfig which gives you all the configs right?

Reply:
yes, this part is right, im making the change now, but do you agree this : `aptos-db-indexer-async-v2` will still be a dependency in storage based on the above reasoning?